lib/ValueAnonymizers/Email.php

```diff
@@ -27,6 +36,13 @@ public function anonymize(Value $value, array $row): AnonymizedValue
 
         $string = $value->getUnEscapedValue();
 
+        if (mb_strpos($string, '@') === false) {
+            throw new FallbackException(
+                new FreeText(new StringHashSha256(new HashAnonymizer())),
+                'The given string is not an email'
+            );
```

Comment:
For simplicity, I think we can have it just like we have the fallback in https://github.com/PayU/mysql-dump-anonymizer/blob/69ee81ed14efa0c34fbbdcfbedefbaaa7f7716ec/lib/ValueAnonymizers/Eav.php#L44

I think the logic for fallback should not leak into the Anonymizer, if we don't necessarily need to.

Suggestion:
```suggestion
            return (new FreeText(new StringHashSha256(new HashAnonymizer())))->anonymize($value, $row);
```

Reply:
changed email fallback in https://github.com/PayU/mysql-dump-anonymizer/pull/21/commits/6eb19180b823fbb6b33aa311c9fd43e07320e656

Reply:
Looks good